src/order/category/BoolAlg.lean

```diff
@@ -0,0 +1,48 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import data.fintype.order
+import order.category.BoundedDistribLattice
+
+/-!
+# The category of boolean algebras
+
+This defines `BoolAlg`, the category of boolean algebras.
+-/
+
+universes u v
+
+open category_theory
+
+/-- The category of boolean algebras. -/
+def BoolAlg := bundled boolean_algebra
+
+namespace BoolAlg
+
+instance : has_coe_to_sort BoolAlg Type* := bundled.has_coe_to_sort
+instance (X : BoolAlg) : boolean_algebra X := X.str
+
+/-- Construct a bundled `BoolAlg` from a `boolean_algebra`. -/
+def of (α : Type*) [boolean_algebra α] : BoolAlg := bundled.of α
+
+instance : inhabited BoolAlg := ⟨of bool⟩
+
+instance : large_category.{u} BoolAlg :=
+{ hom := λ X Y, bounded_lattice_hom X Y,
```

Comment:
Can you inherit the category structure from `BoundedDistribLattice` here?

Reply:
Works, but it's a bit ugly because `boolean_algebra` doesn't extend `distrib_lattice` but `boolean_algebra.core`.

Reply:
What do you mean? The extension properties of these classes are irrelevant for inheriting categorical structure.

Reply:
I don't understand what you mean by inheriting then. I thought you were referring to `bundled_hom.parent_projection`.